nixos/modules/services/matrix/draupnir.nix

```
@@ -0,0 +1,278 @@
+            type = lib.types.str;
+            defaultText = ''
+              URI to the pantalaimon instance, if configured. Else unset.
+            '';
+            description = ''
+              Base URL of the Matrix homeserver, that provides the Client-Server API.
+              If `services.draupnir.pantalaimon.enable` is `true`, this option will become read only. Configure `services.draupnir.pantalaimon.options.homeserver` instead in that case.
+              The listen address of `pantalaimon` will then become the `homeserverUrl` of `draupnir`.
+            '';
+          } // lib.optionalAttrs cfg.pantalaimon.enable {
+            default = "http://${config.services.pantalaimon-headless.instances."draupnir".listenAddress}:${toString config.services.pantalaimon-headless.instances."draupnir".listenPort}/";
+          };
+          #endregion
+
+          #region Common settings
+          autojoinOnlyIfManager = lib.mkOption {
+            type = lib.types.bool;
+            default = false;
+            description = ''
+              If `true`, the bot will only autojoin rooms if the user is a manager.
+            '';
+          };
+
+          automaticallyRedactForReasons = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ ];
+            description = ''
+              A list of reasons for which the bot will automatically redact messages.
+            '';
+          };
+          managementRoom = lib.mkOption {
+            type = lib.types.str;
+            default = "#moderators:example.org";
+            description = ''
+              The room ID or alias where moderators can use the bot's functionality.
+
+              The bot has no access controls, so anyone in this room can use the bot - secure this room!
+
+              Warning: When using a room alias, make sure the alias used is on the local homeserver!
+              This prevents an issue where the control room becomes undefined when the alias can't be resolved.
+            '';
+          };
+
+          protectedRooms = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ ];
+            example = lib.literalExpression ''
+              [
+                "https://matrix.to/#/#yourroom:example.org"
+                "https://matrix.to/#/#anotherroom:example.org"
+              ]
+            '';
+            description = ''
+              A list of rooms to protect (matrix.to URLs).
+              These can also be configured interactively.
+
+              Note that this option does nothing in Draupnir v2+!
+            '';
+          };
+          #endregion
+        };
+      };
+    };
+    #endregion
+  };
+  #endregion
+
+  #region Service configuration
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !(cfg.pantalaimon.enable && cfg.pantalaimon.passwordFile == null);
+        message = "Specify pantalaimon.passwordFile";
```

```suggestion
        assertion = cfg.pantalaimon.enable -> cfg.pantalaimon.passwordFile != null;
        message = "Specifying services.draupnir.pantalaimon.passwordFile is a requirement to enable Pantalaimon.";
```

I think the *imply* (`->`) operator is more readable here

---

I think this is more confusing to read, personally. I could rewrite it as `assertion = cfg.pantalaimon.enable == true && cfg.pantalaimon.passwordFile != null`? i dont quite remember the semantics for assertions

---

> I think this is more confusing to read, personally. I could rewrite it as `assertion = cfg.pantalaimon.enable == true && cfg.pantalaimon.passwordFile != null`? i dont quite remember the semantics for assertions

But in that case this assertion would always fire if `cfg.pantalaimon.enable` is false, as it would be `false && x`. The logical implication `a -> b` is equivalent to `!a || b`.

---

Oh, so it's basically "Check condition B if condition A is true"?

---

Yup